controllers/provisioners/eks/helpers.go

```diff
@@ -156,9 +162,20 @@ func (ctx *EksInstanceGroupContext) GetLabelList() []string {
 	}
 	sort.Strings(labelList)
 
-	// add role label
-	for _, label := range RoleLabelsFmt {
-		labelList = append(labelList, fmt.Sprintf(label, instanceGroup.GetName()))
+	// add role label according to the cluster's k8s version
+	addOldStyleLabel := false
+	clusterVersion, err := ctx.DiscoveredState.GetClusterVersion()
```

Comment:
So if we make `GetClusterVersion` not return an error as mentioned above, we can move the `semver.NewVersion` here.
Also, can we do the opposite nil check? e.g.` err != nil`, this will remove the if nesting.

Reply:
If we fail to `DescribeCluster`, we will get an error during CloudDiscovery.
Otherwise, we will have the cluster version - for this reason we don't need to have the getter return an error, just the value from the DiscoveredState.

In the case of error for semver.NewVersion, does it mean that the version provided is invalid or possibly empty? in this case we should log an error, and fall back to some default.
I think since the "default" behavior is adding both labels, that's what we should do, WDYT? it might mean that nodes don't join the version is actually 1.16+, but in this case, the next reconcile will remove the label.

Reply:
Sorry, I deleted the comment without noticing you had already replayed. I deleted it because you mentioned in a comment below that there is not need to handle the errors from semver, that we could just not add the "Old style role label" if there is an error.

> we don't really need to handle the case of an error here, just don't append anything..
we don't really expect an error to occur here, it would mean there is no cluster version populated?

I agree with logging the errors from semver, although I think we should just add the "New style role label" if there is an error. I think this would potentially allow nodes to join regardless the version of the cluster's k8s, while having a role label. Let me know your final thought and I will implement that.

Reply:
That's fair.. 
We have two options in this case (cannot determine cluster version):
- Add both labels - if cluster is 1.16+, nodes will not join, otherwise same behavior
- Add new label only - if cluster is 1.15-, it might lose a label that was there previously, otherwise no same behavior
So the question is, what's worse - your nodes coming up with a different label (you might be relying on it in some way), or the nodes not joining at all?

In my opinion having the nodes not join will force a human to debug, otherwise, when the cluster version is discoverable in the next reconciles, it will add the correct labels and rotate the nodes so that they can join.
When we allow the nodes to join with different label we change the expected behavior. 
Keep in mind that when the nodes fail to join due to this, it will keep trying to discover the version and when it does, it will do the right thing and add the correct labels.
The end result is, we only want the nodes to join if they have the correct labels.
I'm happy to discuss further if you have a different opinion 👍

Reply:
I understand now, I guess I was underestimating the importance of having the expected labels. Thank you very much.